src/MongoDB.Driver.Core/Core/Authentication/DefaultAuthenticator.cs

```diff
@@ -114,7 +114,10 @@ public async Task AuthenticateAsync(IConnection connection, ConnectionDescriptio
         /// <inheritdoc/>
         public BsonDocument CustomizeInitialIsMasterCommand(BsonDocument isMasterCommand)
         {
-            return isMasterCommand.Merge(CreateSaslSupportedMechsRequest(_credential.Source, _credential.Username)); 
+            isMasterCommand = isMasterCommand.Merge(
+                CreateSaslSupportedMechsRequest(_credential.Source, _credential.Username));
+            return new ScramSha256Authenticator(_credential, _randomStringGenerator)
```

Comment:
Default to SCRAM-SHA-256 when the user doesn't provide us a mechanism
https://github.com/mongodb/specifications/commit/e56724882303a07cb1e836a7a660f28d36af7161#diff-eea9ff495f09b30078faf7f4835e2f14R261

Reply:
Resolving outdated thread